libcontainer/configs/validate/validator.go

```diff
@@ -169,11 +169,22 @@ func (v *ConfigValidator) sysctl(config *configs.Config) error {
 
 func (v *ConfigValidator) intelrdt(config *configs.Config) error {
 	if config.IntelRdt != nil {
-		if !intelrdt.IsEnabled() {
-			return fmt.Errorf("intelRdt is specified in config, but Intel RDT feature is not supported or enabled")
+		if !intelrdt.IsCatEnabled() && !intelrdt.IsMbaEnabled() {
+			return fmt.Errorf("intelRdt is specified in config, but Intel RDT is not supported or enabled")
 		}
-		if config.IntelRdt.L3CacheSchema == "" {
-			return fmt.Errorf("intelRdt is specified in config, but intelRdt.l3CacheSchema is empty")
+
+		if !intelrdt.IsCatEnabled() && config.IntelRdt.L3CacheSchema != "" {
+			return fmt.Errorf("intelRdt.l3CacheSchema is specified in config, but Intel RDT/CAT is not enabled")
+		}
+		if !intelrdt.IsMbaEnabled() && config.IntelRdt.MemBwSchema != "" {
+			return fmt.Errorf("intelRdt.memBwSchema is specified in config, but Intel RDT/MBA is not enabled")
+		}
+
+		if intelrdt.IsCatEnabled() && config.IntelRdt.L3CacheSchema == "" {
```

Comment:
@crosbymichael 
> why is this an issue?

In my opinion, if we specify "intelRdt" in config, we'd better explicitly specify a non-empty string for sub-item "l3CacheSchema" or "memBwSchema".

For example, if we specify the Intel RDT config with empty strings like this: 
```
"linux": {
	"intelRdt": {
		"l3CacheSchema": "",
		"memBwSchema": ""
	}
}
```
When we launch container "container_id", the L3 cache schema and memory bandwidth schema in Intel RDT group "container_id" will keep default values which depend on hardware configuration. This may lead to confusion to user:
```
# cat /sys/fs/resctrl/container_id/schemata
    L3:0=7ff;1=7ff
    MB:0=100;1=100
```

Reply:
ok, thanks for the explanation